datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -402,25 +403,55 @@ def in_developer_mode(self):
         self._log_deprecation('in_developer_mode')
         return False
 
+    def find_typos_in_options(self, user_config, models_config, level):
+        # only import it when running in python 3
+        from jellyfish import jaro_winkler_similarity
+
+        user_config = user_config or {}
+        models_config = models_config or {}
+        typos = set()  # type: Set[str]
+
+        known_options = dict(models_config).keys()
+        unknown_options = sorted(list(user_config.keys() - known_options))
+
+        for unknown_option in unknown_options:
+            similar_known_options = []  # type: List[str]
+            for known_option in known_options:
+                ratio = jaro_winkler_similarity(unknown_option, known_option)
+                if ratio > TYPO_SIMILARITY_THRESHOLD:
+                    similar_known_options = similar_known_options + [known_option]
+                    typos.add(unknown_option)
+
+            if len(similar_known_options) > 0:
+                message = (
+                    'Detected potential typo in configuration option in {}/{} section: `{}`. Did you mean {}?'
+                ).format(self.name, level, unknown_option, ', or '.join(similar_known_options))
+                self.log.warning(message)
+        return typos
```

Comment:
The return value is not used by the callers (outside tests), and this function already did the user visible part of the work. Would it make sense to remove it to make the code a bit simpler?

Reply:
Sure, makes sense!